duecredit/tests/test_collector.py

```diff
@@ -112,126 +107,126 @@ def test_dcite_method():
         @due.dcite("XXX0", path='method')
         def method(arg1, kwarg2="blah"):
             """docstring"""
-            assert_equal(arg1, "magical")
-            assert_equal(kwarg2, 1)
+            assert arg1 == "magical"
+            assert kwarg2 == 1
             return "load"
 
         class SomeClass(object):
             @due.dcite("XXX0", path='someclass:method')
             def method(self, arg1, kwarg2="blah"):
                 """docstring"""
-                assert_equal(arg1, "magical")
-                assert_equal(kwarg2, 1)
+                assert arg1 == "magical"
+                assert kwarg2 == 1
                 return "load"
 
         if active:
-            assert_equal(due.citations, {})
-            assert_equal(len(due._entries), 1)
+            assert due.citations == {}
+            assert len(due._entries) == 1
 
-        yield _test_dcite_basic, due, method
+        _test_dcite_basic(due, method)
 
         if active:
-            assert_equal(len(due.citations), 1)
-            assert_equal(len(due._entries), 1)
+            assert len(due.citations) == 1
+            assert len(due._entries) == 1
             citation = due.citations[("method", "XXX0")]
-            assert_equal(citation.count, 1)
+            assert citation.count == 1
             # TODO: this is probably incomplete path but unlikely we would know
             # any better
-            assert_equal(citation.path, "method")
+            assert citation.path == "method"
 
         instance = SomeClass()
-        yield _test_dcite_basic, due, instance.method
+        _test_dcite_basic(due, instance.method)
 
         if active:
-            assert_equal(len(due.citations), 2)
-            assert_equal(len(due._entries), 1)
+            assert len(due.citations) == 2
+            assert len(due._entries) == 1
             # TODO: we should actually get path/counts pairs so here
             citation = due.citations[("someclass:method", "XXX0")]
-            assert_equal(citation.path, "someclass:method")
-            assert_equal(citation.count, 1)
+            assert citation.path == "someclass:method"
+            assert citation.count == 1
 
             # And we explicitly stated that module need to be cited
-            assert_true(citation.cite_module)
-
+            assert citation.cite_module
 
         class SomeClass2(object):
```

Comment:
please add a comment that this class is not directly used and just there to test for that ;)